### ISSUE TITLE
differences between original repo and this fork

### ISSUE THREAD

**Issue by lyzband** [AlexeyAB/darknet#705]
I have trained and detected with my own data set using original repo(pjreddie darknet) and this fork one after another under exactly same condition but they showed different results.
sometimes this fork double-detects the objects as you can see from the following image, while it never happens with original repo.

I followed the comment (#232) to make the results identical but still not perfectly matching, the double-detecting problem got relatively solved though.
And then I figured that the code not only [ here](https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L1072-L1073) but also there:
- original repo
https://github.com/pjreddie/darknet/blob/80d9bec20f0a44ab07616215c6eadb2d633492fe/examples/detector.c#L64-L65

- this fork
https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L119-L120

 are different.
What does this difference mean?
Will I get the identicla result if I change this line?
Looking forward to hearing from you

**Comment by AlexeyAB** [comment]
Hi, 

> original repo
> https://github.com/pjreddie/darknet/blob/80d9bec20f0a44ab07616215c6eadb2d633492fe/examples/detector.c#L64-L65
> 

You refer to the old commit from original repo where was `nms = 0.3` https://github.com/pjreddie/darknet/blob/80d9bec20f0a44ab07616215c6eadb2d633492fe/examples/detector.c#L579

----

Try to use the same `nms = 0.45` in both repo.

* https://github.com/AlexeyAB/darknet/blame/master/src/detector.c#L1057
* https://github.com/pjreddie/darknet/blame/master/examples/detector.c#L575

----

Also try to use the same weights & cfg files on this repo and original repo.

**Comment by lyzband** [comment]
oh, I refered to wrong commit by mistake, however, I used both repos with same nms( = .45) while testing and i also have tried to use the same weights and cfgs but still got the different results.
that's why i have changed the code in detection part of this repo(#232 ), to keep same ratio while resizing, but the results from each repo were still different(even with same nms, weight, and cfg).

and then I found the different lines from 2 repos, as i mentioned earlier: 

- https://github.com/pjreddie/darknet/blob/master/examples/detector.c#L65-L66
- https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L119-L120

Anyway, to be more clearer, my goal is to get the exactly identical results from original repo and this repo using same weight and cfg.
Any other lines in this repo that I have to change for that?

**Comment by AlexeyAB** [comment]
> and then I found the different lines from 2 repos, as i mentioned earlier:
> 
> https://github.com/pjreddie/darknet/blob/master/examples/detector.c#L65-L66
> https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L119-L120

Yes, it affects only on training. But if you use the same cfg/weights for detection, it should not affect.

----

1. Check that you use letterbox_image: https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/src/detector.c#L1072-L1073

----

2. Also there was fixed [reorg] layer: `fixed usage of [reorg]-layer` https://github.com/AlexeyAB/darknet/issues/529#issuecomment-377204382

It was done because: (1) https://github.com/pjreddie/darknet/issues/344 and (2) https://github.com/opencv/opencv/pull/9705#discussion_r143136536


* Or (for this repository) use `[reorg_old]` instead of `[reorg]` in your yolo-voc.cfg or yolo.cfg
* Or undo these changes: https://github.com/AlexeyAB/darknet/commit/fe44d3d0f29e96714de75234e1fd7e40b05b0296

----

3. Also there is fused Conv-layer and batch-norm layer to speedup detection about 7%.
Just comment this line: https://github.com/AlexeyAB/darknet/blob/5e3dcb6f34868e341466b57b13ad63f86b337250/src/detector.c#L1051

**Comment by lyzband** [comment]
>1. Check that you use letterbox_image:
darknet/src/detector.c
Lines 1072 to 1073 in 5e3dcb6
       //image sized = resize_image(im, net.w, net.h); 
 image sized = letterbox_image(im, net.w, net.h); letterbox = 1; 

Yeap I have done it already.

> 2. Also there was fixed [reorg] layer: fixed usage of [reorg]-layer #529 (comment)
It was done because: (1) pjreddie#344 and (2) opencv/opencv#9705 (comment)
-Or (for this repository) use [reorg_old] instead of [reorg] in your yolo-voc.cfg or yolo.cfg
-Or undo these changes: fe44d3d

I am using my own weights which were trained from yolov3 via original repo.
And it seems like yolov3.cfg does not contain [reorg] layer anymore.

> 3. Also there is fused Conv-layer and batch-norm layer to speedup detection about 7%.
Just comment this line:
darknet/src/detector.c
Line 1051 in 5e3dcb6
 fuse_conv_batchnorm(net); 

this change is for speed up so it does not affect the detection result.

In conclusion, the only option that I could apply is to change 'resize_image' into 'letterbox_image' but that is what I have already done to my code. Any suggestions?
